zebedee-reader/src/main/java/com/github/onsdigital/zebedee/util/URIUtils.java

```diff
@@ -79,11 +87,94 @@ public static String getLastSegment(String uri) {
         return segments[segments.length - 1];
     }
 
-    public static String removeLastSegment(String uri) {
-        if (isEmpty(uri)) {
-            return null;
+    public static String removeAfterKeyword(String path, String keyword) {
+        int index = path.indexOf(keyword);
+        if (index == -1) {
+            return path;
+        }
+        String result = path.substring(0, index);
+        return result.endsWith("/") ? result.substring(0, result.length() - 1) : result;
+    }
+
+    public static String getQueryParameterFromURL(String url, String queryParam) throws UnsupportedEncodingException {
```

Comment:
Why not catch `UnsupportedEncodingException` instead of throwing it? You're already catching `URISyntaxException`, is there a reason why you don't want to catch this one too?

Reply:
yes makes sense to catch as close as possible to the origin